### PR TITLE
Add type annotations for built-in functions

### PR DESCRIPTION
**Pull request by xuebinsu**
SQL has several built-in functions that are widely used. For example,
count(), avg(), and max(). For better user experience, we add the
corresponding Python functions for them with full type annotations
so that the use will get better suggestions from their IDEs.

### CODE REVIEW

File: greenplumpython/table.py [hunk 1]
```diff
@@ -576,7 +576,7 @@ def apply(self, func: Callable[["Table"], "FunctionExpr"]) -> "FunctionExpr":
         """
         Apply a function to the table
         """
-        return func(self)
+        return func(self)(table=self)
```
Comment (Quasimewdo) [inline greenplumpython/table.py:579]: Why add the second parenthesis?

**Review by yihong0618 — APPROVED** [review]
LGTM

File: greenplumpython/table.py (hunk 1 above)
Comment (Quasimewdo) [reply greenplumpython/table.py:586]: Got it, thanx

**Review by Quasimewdo — APPROVED** [review]
+1

**Pull request merged**